### ISSUE TITLE
[Bug]: Unable to read file: No file system handle registered (\home)

### ISSUE THREAD

**Issue by jeengbe** [coder/code-server#5736]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### OS/Web Information

- Web Browser: Chrome 108.0.5359.22 (Official Build) beta (64-bit) (cohort: Beta) 
- Local OS: Windows 11 Version 22H2 (Build 22621.755)
- Remote OS: Debian GNU/Linux 11 (bullseye)
- Remote Architecture: x86_64
- `code-server --version`: 1.72.1 129500ee4c8ab7263461ffe327268ba56b9f210d x64

(Also tested on WSL2 Ubuntu 20.04.5 LTS (Focal Fossa) with Chrome 107.0.5304.87 (Official Build) (64-bit))

### Steps to Reproduce

1. open code-server
2. clone [jeengbe/vality](https://github.com/jeengbe/vality)
3. run  `pn i && cd packages/vality && pnpn test`
4. ctrl-click the name of a test file (to open) and see the error

### Expected

As correctly annotated by the hover, the file should open upon ctrl-click.

### Actual

Instead of opening the file, a new tab opens with an error message about the file not being found. 

### Logs

Browser log: 

```log
  ERR Error: Unable to read file '\home\coder\data\vality\packages\vality\tests\validate.test.ts' (Unavailable (FileSystemError): Error: No file system handle registered (\home))
    at c.doReadFileStream (fileService.ts:573:11)
    at async n.doRead (textFileService.ts:219:19)
    at async n.readStream (textFileService.ts:198:35)
    at async Ar.resolveFromFile (textFileEditorModel.ts:433:20)
    at async a.doResolve (textFileEditorModelManager.ts:416:4)
    at async _l.doResolveAsText (fileEditorInput.ts:328:4)
    at async sc.setInput (textFileEditor.ts:99:26)
    at async r.doSetInput (editorPanes.ts:353:4)
    at async r.doOpenEditor (editorPanes.ts:222:34)
    at async r.openEditor (editorPanes.ts:126:11)
```

### Screenshot/Video

How it started
![image](https://user-images.githubusercontent.com/51749023/198826202-aaeb7d4a-e7d1-4c9d-92e2-6af29abddcfc.png)

How it's going
![image](https://user-images.githubusercontent.com/51749023/198826217-1d0dbddc-84b7-48f8-b3c7-95d1e9bc21f2.png)

### Does this issue happen in VS Code or GitHub Codespaces?

- [X] I cannot reproduce this in VS Code.
- [X] I cannot reproduce this in GitHub Codespaces.

### Are you accessing code-server over HTTPS?

- [X] I am using HTTPS.

### Notes

I tried to reproduce it in a simple environment, but I couldn't replicate the issue, hence the link to my actual project.

I don't have Codespaces so I couldn't actually try it there.

**Comment by jsjoeio** [comment]
Hmm...so to cut down the issue to the core, seems like you click on a file path in the terminal and expect it to open in code-server, right?

I tried reproducing that but couldn't on macOS + Brave.

https://user-images.githubusercontent.com/3806031/199092971-81f05dde-44a4-4f95-9ce4-664199756eb5.mov

One difference - when i hover over the file name in the terminal, it doesn't show a path 🤔 

I did try with your repo as well and it worked as expected.
